packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -75,10 +83,17 @@ function preload(
       seen[dep] = true
```

Comment:
This is just an idea but maybe we could simplify the logic if we use `new URL(dep, document.baseURI).href`. If the SSR markup used an absolute path, I guess the current approach won't work. (it didn't work from before though)

Reply:
that was what I missed! To supply the base url!
Earlier I used `new URL(dep).href` that didn't make sense when dep is relative, just understand it now, according to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#parameters) about the constructor signature `new URL(url, base)`:

> If `url` is a relative URL, `base` is required, and will be used as the base URL. If `url` is an absolute URL, a given `base` will be ignored.

Reply:
> This is just an idea but maybe we could simplify the logic if we use `new URL(dep, document.baseURI).href`. If the SSR markup used an absolute path, I guess the current approach won't work. (it didn't work from before though)

NOOOOOOO! BaseURI [isn't supported on IE11](https://caniuse.com/?search=baseuri)!

Reply:
> > This is just an idea but maybe we could simplify the logic if we use `new URL(dep, document.baseURI).href`. If the SSR markup used an absolute path, I guess the current approach won't work. (it didn't work from before though)
> 
> NOOOOOOO! BaseURI [isn't supported on IE11](https://caniuse.com/?search=baseuri)!

OK, since this PR doesn't introduce legacy support, I think it's fine to use BaseURI.
**It seems that a correct place to handle this issue is at my PR #9920, since right now the preload function isn't called for legacy browsers.**
Therefore, @bgoscinski, you may ignore the rest of this comment, I'll handle with it after in my PR.

The full solution was introduced in sapper legacy support, on PR sveltejs/sapper#1562.
I'll explain briefly the logic of this sapper PR, how it handle the situation when `document.baseURI` isn't available:
We could have used the `document.URL` which is supported by IE11. Sadly, due to the reasons in [this stackoverflow answer](https://stackoverflow.com/a/52241756/2968864), if there is some `<base>` element in the DOM, it changes the base path, ignoring the current URL(demo is included in the stackoverflow answer). Luckily, by the specifications, there could be only one at most such `<base>` element, so we just need to grab the one we see, and take this to be the baseURI. If no such tag were found, we can simply take `document.URL` instead.